src/library_wasmfs.js

```diff
@@ -95,11 +107,22 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       mode = typeof mode == 'undefined' ? 438 /* 0666 */ : mode;
       return withStackSave(() => {
         var buffer = stringToUTF8OnStack(path);
-        return __wasmfs_open({{{ to64('buffer') }}}, flags, mode);
+        var fd = __wasmfs_open({{{ to64('buffer') }}}, flags, mode);
+        if(fd < 0) {
+          FS.DoWasmFSError(-fd);
+        }
+        return fd;
       })
     },
     // TODO: create
     // TODO: close
+    close: (fd) => {
+      var err = __wasmfs_close(fd);
+      if(err == {{{ cDefs.EBADF}}}) {
```

Comment:
I don't think we need to check for specific errors - that would have the risk of us forgetting some checks. Just doing `if < 0, error` seems best. In fact, the `< 0` check might be inside the helper function, which could be called `handleError` or `throwIfError`?

Reply:
I considered putting a `< 0` check inside the error function, but sometimes the error codes returned by the functions are positive or negative based on the POSIX API, so would it be better to just continue handling the error checks inside each function?

Reply:
Oh right, good point. But is zero always success, though? If so then maybe calling `handleError(-x)` (negating the sign) could be used in the right places. If not then let's do it as you suggested.

Reply:
Ah, I see what you mean. I changed the implementation to add a helper function called `handleError()` that takes in the return value, and assumes that negative values are errors. This is so I can use the helper with methods like `FS.open()` that return positive values on success and negative values on error. Like you said, I just negate the sign when calling `handleError()` as necessary, like for `FS.close()`.